ui/src/app/rsd/views/RSDList/RSDListTable/RSDListTable.test.tsx

```diff
@@ -0,0 +1,275 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import { MemoryRouter } from "react-router-dom";
+import configureStore from "redux-mock-store";
+import React from "react";
+
+import type { RootState } from "app/store/root/types";
+import {
+  pod as podFactory,
+  podState as podStateFactory,
+  resourcePool as resourcePoolFactory,
+  resourcePoolState as resourcePoolStateFactory,
+  rootState as rootStateFactory,
+  zone as zoneFactory,
+  zoneState as zoneStateFactory,
+} from "testing/factories";
+import RSDListTable from "./RSDListTable";
+
+const mockStore = configureStore();
+
+describe("RSDListTable", () => {
+  let initialState: RootState;
+  beforeEach(() => {
+    const pods = [
+      podFactory({ pool: 1, type: "rsd", zone: 1 }),
+      podFactory({ pool: 2, type: "rsd", zone: 2 }),
+    ];
+    initialState = rootStateFactory({
+      pod: podStateFactory({ items: pods, loaded: true }),
+      resourcepool: resourcePoolStateFactory({
+        loaded: true,
+        items: [
+          resourcePoolFactory({ id: pods[0].pool }),
+          resourcePoolFactory({ id: pods[1].pool }),
+        ],
+      }),
+      zone: zoneStateFactory({
+        loaded: true,
+        items: [
+          zoneFactory({ id: pods[0].zone }),
+          zoneFactory({ id: pods[1].zone }),
+        ],
+      }),
+    });
+  });
+
+  it("correctly fetches the necessary data", () => {
+    const state = { ...initialState };
+    const store = mockStore(state);
+    mount(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={[{ pathname: "/rsd", key: "testKey" }]}>
+          <RSDListTable />
+        </MemoryRouter>
+      </Provider>
+    );
+    const expectedActions = ["pod/fetch", "resourcepool/fetch", "zone/fetch"];
+    const actualActions = store.getActions();
+    expect(
+      actualActions.every((action) => expectedActions.includes(action.type))
+    ).toBe(true);
+  });
+
+  it("shows RSDs sorted by descending FQDN by default", () => {
+    const state = { ...initialState };
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={[{ pathname: "/rsd", key: "testKey" }]}>
+          <RSDListTable />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find('[data-test="fqdn-header"] i').exists()).toBe(true);
+    expect(
+      wrapper
+        .find('[data-test="fqdn-header"] i')
+        .props()
+        .className?.includes("u-mirror--y")
```

Comment:
Another case where I'm not sure if the `?` guard is necessary. Is this make the tests fail in a nicer way?

Reply:
This one is to satisfy strict mode, otherwise I get an `object could be undefined` error. :man_shrugging: